packages/react-dom/src/__tests__/ReactUpdates-test.js

```diff
@@ -1100,212 +1249,110 @@ describe('ReactUpdates', () => {
     }
 
     const container = document.createElement('div');
+    const root = ReactDOMClient.createRoot(container);
     function render() {
-      ReactDOM.render(
+      root.render(
         <Editor
           onChange={newProps => {
+            onChangeCalled = true;
             props = {...props, ...newProps};
             render();
           }}
           {...props}
         />,
-        container,
       );
     }
 
     let props = {text: 'hello', rendered: false};
-    render();
+    await act(() => {
+      render();
+    });
     props = {...props, text: 'goodbye'};
-    render();
+    await act(() => {
+      render();
+    });
     expect(container.textContent).toBe('goodbye');
     expect(mounts).toBe(1);
+    expect(onChangeCalled).toBeTruthy();
```

Comment:
Added this because I was worried the change handler wasn't being called, which I think would still pass the above assertions.